pkg/server/server_test.go

```diff
@@ -36,3 +37,41 @@ func (s *TestServerSuite) TestServer() {
 	// Check that Engine() returns the router object.
 	require.NotNil(s.T(), srv.Engine())
 }
+
+func (s *TestServerSuite) TestCrossCors() {
+	// We're using the example config for the configuration here as the
+	// specific config params do not matter for testing the routes setup.
+	srv := server.New(s.Config(), fake.NewMockableApplication(s.Config(), nil))
+
+	// Setting up the routes.
+	err := srv.SetupRoutes()
+	require.NoError(s.T(), err)
+
+	// Check that there are routes registered.
+	routes := srv.GetRegisteredRoutes()
+	require.NotEmpty(s.T(), routes)
+
+	// Check that Engine() returns the router object.
+	require.NotNil(s.T(), srv.Engine())
+
+	go srv.Engine().Run()
```

Comment:
Shouldn't we wait for the server to be available or something before proceeding with the tests?

Reply:
I don't think so: https://github.com/gin-gonic/gin/blob/master/gin.go#L304

Reply:
You are starting a new goroutine which will be blocked. But the test goroutine will proceed. This is exactly why you need to start a new goroutine here so the test is not blocked. But it takes some time to start the server and it might happen then when you make that OPTIONS request the server is not started yet.